src/object/schedcontext.c

```diff
@@ -401,7 +421,9 @@ void schedContext_cancelYieldTo(tcb_t *tcb)
 void schedContext_completeYieldTo(tcb_t *yielder)
 {
     if (yielder && yielder->tcbYieldTo) {
-        setConsumed(yielder->tcbYieldTo, lookupIPCBuffer(true, yielder));
+        /* FIXME: this should only be true here if the original
+                  invocation we are completing was Call. */
+        setConsumed(yielder->tcbYieldTo, yielder, true);
```

Comment:
It's a complete hack, but you could set yielder to tcbYieldTo to indicate you don't want to reply to the yielder.

Reply:
Sorry for the confusion. I meant setting `scYieldFrom` equal to `tcbYieldTo` instead of the actual yielder in `invokeSchedContext_YieldTo`. This could be done because you don't need to remember who the actual yielder was in case of a non-call.

Leaving it as-is is fine of course.

Reply:
I'll turn this one into an issue and work on it when I'm back from travel. Michael is blocked on getting this PR merged and the current state of the PR makes the current behaviour strictly better, even if it doesn't fully fix this specific issue, so I think we should go ahead.

Reply:
See https://github.com/seL4/seL4/issues/1471